### DISCUSSION TITLE
getting stress_old in ComputeLinearElasticPFFractureStress

### DISCUSSION THREAD

**Discussion by selarem** [idaholab/moose#16552] [Q&A Modules: Phase field] [answered]
Hi all,
I am trying to get _stress_old in ComputeLinearElasticPFFractureStress by writting at the constructor:

_stress_old(getMaterialPropertyOld<RankTwoTensor>(_base_name + "stress"))

but when compiling I obtain:

projects/moose/modules/tensor_mechanics/src/materials/ComputeLinearElasticPFFractureStress.C:34:5: error: class 'ComputeLinearElasticPFFractureStress' does not have any field named '_stress_old'
     _stress_old(getMaterialPropertyOld<RankTwoTensor>(_base_name + "stress"))

Thanks for any suggestion.

Regards

**Comment by aeslaughter** [comment, marked as answer]
`_stress_old` needs to be defined in the header file with the correct type. The call to getMaterialPropertyOld should also contain a template argument: `getMaterialPropertyStressOld<TheTypeHere>(...)`. 

Finally, are you modifying this class and plan to push the changes to MOOSE? Otherwise, you should create your own Material object that inherits from this within you application or a separate Material object that uses the properties from this Material.

**Reply by rh201** [reply]
@aeslaughter Hi! What's the correct type do you refer to? I searched **getMaterialPropertyStressOld**  but found no results. Are there any examples of that? 

Also, when I modify **getMaterialPropertyOld** to **getADMaterialPropertyOld**, it seems not work, so does AD contain the function of **getADMaterialPropertyOld** ? Thank you!

**Reply by hugary1995** [reply]
Can you tell us why you want to get stress_old? This seems to be an unusual request, But if you describe what you want to achieve here, we might be able to suggest you the right thing to do here.

**Reply by rh201** [reply]
Hi Gary, what I want is strain_old to calculate the strain rate. Maybe the question #18822 is much clear.
